Source/UnitEx.cpp

```diff
@@ -188,8 +223,12 @@ enum OP2UnitFlags
 	flagVehicle = 0x2,
 	flagBuilding = 0x4,
 	flagDoubleFireRate = 0x20,
+	flagPower = 0x2000,
+	flagWorkers = 0x4000,
+	flagScientists = 0x8000,
 	flagLive = 0x20000,
 	flagStickyfoamed = 0x40000,
+	flagInfected = 0x40000,

```

Comment:
This looks like a possible typo or mistake. The Stickyfoamed and Infected flags are the same value.

Reply:
I don't think this is a mistake. It appears to be context sensitive. Ex. If building, the flag stands for "infected", if vehicle, the flag stands for "stickyfoamed". I remember checking this by blighting my spaceport, but it's been quite a few months now. I never did verify "stickyfoamed".

Reply:
This matches with the text files describing the `Unit` class (in the OllyDbg folder in the SVN). The flag is double use, as TechCor says, where meaning is context dependent on either building or vehicle. Quite a number of flags are like that.

Reply:
Sounds good. I tried testing the functions:

```C++
int UnitEx::IsEMPedEx()
int UnitEx::IsStickyfoamed()
int UnitEx::IsESGed()
```

By implementing the code below. All 3 functions did not register properly when the unit was ESGed, EMPed, or Stickyfoamed. Maybe I'm doing something wrong. I'll write an issue against it. The function Unit::IsEMPed worked properly, so I'm not actually sure why the IsEMPedEx function was writted?

```C++
ExportLevelDetailsFullEx("Test Colony Game", "cm02.map", "MULTITEK.TXT", MissionTypes::Colony, 2, 12, false, 1);

UnitEx unit;

Export int InitProc()
{
	TethysGameEx::CreateUnit(unit, mapLynx, LOCATION(50, 50), 1, mapESG, South);
	TethysGameEx::CreateUnit(unit, mapCargoTruck, LOCATION(50, 50), 0, mapNone, South);
	return true;
}

Export void AIProc() 
{
	if (unit.IsEMPedEx() ||
		unit.IsStickyfoamed() ||
		unit.IsESGed()) 
	{
		TethysGame::AddGameSound(SoundID::sndAcid2, 0);
	}
}
```

Reply:
All the functions depend on HFL being initialized. Is there perhaps a missing initialization? I've run into that problem before, where HFL functions seemingly do nothing, without indication of a problem, and it's because of the initialization checks.

Looks like the `IsEMPedEx()` method actually gets the EMP time. The method could be named better. Same with `IsStickyfoamed()` and `IsESGed()`.

Anyway, those methods were pre-existing.